### DISCUSSION TITLE
No docs on "inner" middleware or "useInnerOffset" Interaction?

### DISCUSSION THREAD

**Discussion by Abyrd9** [floating-ui/floating-ui#1953] [Q&A]
On the [MacIOs Select example](https://codesandbox.io/s/shy-snowflake-kp6479?file=/src/Select.tsx), it makes use of an "inner" middleware that has methods like `onFallbackChange` but there's nothing in the docs it seems about any of this. Nor about the hook `useInnerOffset`. Can someone explain these?

**Comment by atomiks** [comment]
These are currently missing docs 😞

**Reply by atomiks** [reply]
Added docs: https://floating-ui.com/docs/inner

**Reply by Abyrd9** [reply]
WOW! Thanks for adding that same day 😂 I am impressed and grateful 🙏🏻
